index.js

```diff
@@ -14,18 +17,64 @@ module.exports = {
   },
 
   setupPreprocessorRegistry: function(type, registry) {
-    if (type !== 'self') return;
+    this.setupJsPreprocessing(registry, type);
+
+    if (type === 'self') {
```

Comment:
This may be unrelated but does that mean that if I were building an app that used this addon, I could not define a `myapp/app/styles/components/ui-button--foo.scss` to generate the component?